PluginsAndFeatures/azure-toolkit-for-intellij/azure-intellij-plugin-azure-sdk/src/test/java/com/microsoft/azure/toolkit/intellij/azure/sdk/buildtool/HardcodedAPIKeysAndTokensCheckTest.java

```diff
@@ -0,0 +1,134 @@
+ These are some instances that a flag would be raised.
+ * 1. TextAnalyticsClient client = new TextAnalyticsClientBuilder()
+ *         .endpoint(endpoint)
+ *         .credential(new AzureKeyCredential(apiKey))
+ *         .buildClient();
+ *
+ * 2. TokenCredential credential = request -> {
+ *         AccessToken token = new AccessToken("<your-hardcoded-token>", OffsetDateTime.now().plusHours(1));
+ *     }
+ */
+public class HardcodedAPIKeysAndTokensCheckTest {
+
+    // Declare as instance variables
+    @Mock
+    private ProblemsHolder mockHolder;
+
+    @Mock
+    private PsiElementVisitor mockVisitor;
+
+    @BeforeEach
+    public void setup() {
+        mockHolder = mock(ProblemsHolder.class);
+        mockVisitor = createVisitor();
+    }
+
+
+    /**
+     * Test the HardcodedAPIKeysAndTokensCheck class for hardcoded API keys and tokens.
+     * When a client is authenticated with AzurekeyCredentials and AccessToken, a problem is registered.
+     */
+    @Test
+    public void testHardcodedAPIKeysAndTokensCheck() {
+
+        // assert visitor
+        assertVisitor(mockVisitor);
+
+        int NUMBER_OF_INVOCATIONS = 1; // number of times registerProblem is called
+
+        String CLIENT_TO_CHECK_ONE = "AccessToken";
+        verifyRegisterProblem(CLIENT_TO_CHECK_ONE, NUMBER_OF_INVOCATIONS);
+
+        String CLIENT_TO_CHECK_TWO = "AzureKeyCredential";
+        verifyRegisterProblem(CLIENT_TO_CHECK_TWO, NUMBER_OF_INVOCATIONS);
+    }
+
+    /**
+     * Test for non-auth client use.
+     * These are other Azure clients that are not AzureKeyCredential or AccessToken.
+     */
+    @Test
+    public void testNonAuthAzureClientUse() {
+
+        // assert visitor
+        assertVisitor(mockVisitor);
+
+        String CLIENT_TO_CHECK = "SomeOtherClient";
+        int NUMBER_OF_INVOCATIONS = 0;
+        verifyRegisterProblem(CLIENT_TO_CHECK, NUMBER_OF_INVOCATIONS);
+    }
+
+    /**
+     * Test for null class reference.
+     */
+    @Test
+    public void testNullClassReference() {
+
+        // assert visitor
+        assertVisitor(mockVisitor);
+
+        String CLIENT_TO_CHECK = ""; // null class reference
+        int NUMBER_OF_INVOCATIONS = 0;
+        verifyRegisterProblem(CLIENT_TO_CHECK, NUMBER_OF_INVOCATIONS);
+    }
+
+
+    // Helper method to create visitor.
+    PsiElementVisitor createVisitor() {
+        boolean isOnTheFly = true;
+        APIKeysAndTokensVisitor visitor = new HardcodedAPIKeysAndTokensCheck.APIKeysAndTokensVisitor(mockHolder, isOnTheFly);
+        return visitor;
+    }
+
+
+    // Helper method to assert visitor is not null and is an instance of JavaElementVisitor
+    void assertVisitor(PsiElementVisitor mockVisitor) {
+
+        assertNotNull(mockVisitor);
+        assertTrue(mockVisitor instanceof JavaElementVisitor);
+    }
+
+    // Helper method to verify registerProblem is called
+    void verifyRegisterProblem(String CLIENT_TO_CHECK, int NUMBER_OF_INVOCATIONS) {
```

Comment:
do we need these methods package-private

Reply:
oh they should be private actually thanks:)